src/Aspire.Hosting.Azure/IAddRoleAssignmentsContext.cs

```diff
@@ -38,4 +38,9 @@ public interface IAddRoleAssignmentsContext
     /// Not all role assignments require/use a principal name.
     /// </remarks>
     public BicepValue<string> PrincipalName { get; }
+
+    /// <summary>
+    /// Gets the client ID to use in the role assignment.
+    /// </summary>
+    public BicepValue<string> ClientId { get; }
```

Comment:
This is a breaking API change but we don't really care 😄

Reply:
Yeah, this interface was added in 9.2, and I don't expect anyone else to implement it.

If we care, we could use a DIM here and throw NotImplementedException.